sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/ConfidentialLedgerFailoverService.cs

```diff
@@ -0,0 +1,262 @@
+            try
+            {
+                using JsonDocument jsonDoc = JsonDocument.Parse(response.Content);
+                jsonDoc.RootElement.TryGetProperty("ledgerId", out _); // optional
+                if (jsonDoc.RootElement.TryGetProperty("failoverLedgers", out JsonElement failoverArray))
+                {
+                    foreach (JsonElement failoverLedger in failoverArray.EnumerateArray())
+                    {
+                        string failoverLedgerId = null;
+                        try
+                        {
+                            switch (failoverLedger.ValueKind)
+                            {
+                                case JsonValueKind.String:
+                                    failoverLedgerId = failoverLedger.GetString();
+                                    break;
+                                case JsonValueKind.Object:
+                                    if (failoverLedger.TryGetProperty("name", out JsonElement nameProp) && nameProp.ValueKind == JsonValueKind.String)
+                                    {
+                                        failoverLedgerId = nameProp.GetString();
+                                    }
+                                    else
+                                    {
+                                        foreach (JsonProperty prop in failoverLedger.EnumerateObject())
+                                        {
+                                            if (prop.Value.ValueKind == JsonValueKind.String && string.Equals(prop.Name, "id", StringComparison.OrdinalIgnoreCase))
+                                            {
+                                                failoverLedgerId = prop.Value.GetString();
+                                                break;
+                                            }
+                                        }
+                                    }
+                                    break;
+                            }
+                        }
+                        catch (JsonException jex)
+                        {
+#if DEBUG
+                            Debug.WriteLine($"[ConfidentialLedgerFailoverService] JSON parse issue for failoverLedger element: {jex.Message}");
+#endif
+                            _ = jex; // suppress unused warning in non-DEBUG builds
+                        }
+                        catch (InvalidOperationException ioex)
+                        {
+#if DEBUG
+                            Debug.WriteLine($"[ConfidentialLedgerFailoverService] Invalid operation while parsing failoverLedger element: {ioex.Message}");
+#endif
+                            _ = ioex; // suppress unused warning in non-DEBUG builds
+                        }
+
+                        if (!string.IsNullOrEmpty(failoverLedgerId))
+                        {
+                            Uri endpoint = new UriBuilder(primaryEndpoint) { Host = $"{failoverLedgerId}.{LedgerDomainSuffix}" }.Uri;
```

Comment:
primary endpoint host already contains the suffix, which might be staging or any new env we add later like ppe. Therefore if would make sense to replace the first part of the existing host instead of constructing the new one with the current prod suffix.